src/borg/repository.py

```diff
@@ -493,9 +493,23 @@ def open(self, path, exclusive, lock_wait=None, lock=True):
         self.id = unhexlify(self.config.get("repository", "id").strip())
         self.io = LoggedIO(self.path, self.max_segment_size, self.segments_per_dir)
 
+    def _load_hints(self):
+        if not (transaction_id := self.get_transaction_id()):
+            # self is a fresh repo, so transaction_id is None and there is no hints file
```

Comment:
maybe it works like this (not sure if we would ever get a transaction_id of 0), but still, if you want to check for just `None`, please use `... is None` and not `not ...`.